nipype/interfaces/dtitk/utils.py

```diff
@@ -87,46 +81,26 @@ class TVAdjustVoxSpTask(CommandLineDtitk):
     input_spec = TVAdjustVoxSpInputSpec
     output_spec = TVAdjustVoxSpOutputSpec
     _cmd = 'TVAdjustVoxelspace'
-    _suffix = '_reslice'
-
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.out_file
-        if not isdefined(self.inputs.out_file):
-            outputs["out_file"] = self._gen_fname(self.inputs.in_file,
-                                                  suffix=self._suffix,
-                                                  ext='.'+'.'.join(
-                                                      self.inputs.in_file.
-                                                      split(".")[1:]))
-        outputs["out_file"] = os.path.abspath(outputs["out_file"])
-        return outputs
-
-    def _gen_filename(self, name):
-        if name == "out_file":
-            return self._list_outputs()["out_file"]
-        return None
 
 
 # TODO not using these yet... need to be tested
 
-
 class SVAdjustVoxSpInputSpec(CommandLineInputSpec):
-    in_file = traits.Str(desc="image to resample", exists=True,
+    in_file = File(desc="image to resample", exists=True,
                          mandatory=True, position=0, argstr="-in %s")
-    in_target = traits.Str(desc='target volume', exists=True, mandatory=False,
+    in_target = File(desc='target volume', mandatory=True,
                            position=2, argstr="-target %s")
-    in_voxsz = traits.Str(desc='resampled voxel size', exists=True,
-                          mandatory=False, position=3, argstr="-vsize %s")
-    out_file = traits.Str(desc='output path', exists=True, mandatory=False,
-                          position=1, argstr="-out %s",
-                          name_source="in_file",
-                          name_template='%s_origmvd.nii.gz')
-    origin = traits.Str(desc='xyz voxel size', exists=True, mandatory=False,
+    in_voxsz = traits.Str(desc='resampled voxel size', mandatory=True,
+                           position=3, argstr="-vsize %s")
+    out_file = traits.Str(desc='output path', position=1, argstr="-out %s",
+                          name_source="in_file", name_template='%s_reslice',
+                          keep_extension=True)
+    origin = traits.Str(desc='xyz voxel size', mandatory=True,
```

Comment:
I'm going to set the default of this to be (0,0,0) and set usedefault=True because that is recommended by the tutorial. If a target image is provided, this will be ignored anyway so I think it's fine to do. Let me know if you disagree.

Reply:
I'm personally disinclined to add arguments that will be ignored, since that could be confusing to someone attempting to understand or debug a pipeline. You could set it to `xor=['in_target']`, to make clear that these are mutually exclusive options.

Also, I think you probably want to use numeric data type here, rather than a string:

```Python
origin = traits.Tuple(traits.Float, traits.Float, traits.Float, mandatory=True,
                      xor=['in_target'], position=4, argstr='-origin (%g,%g,%g)')
```

Similarly, `in_target` should list `origin` in its `xor` parameter. (Mandatory is okay in mutually exclusive cases.)

Reply:
There are multiple parameters that will be ignored if 'in_target' is provided. You can either provide a target image to match parameters to (origin & voxel size) or give parameters individually, but you do not need to provide both. E.g. one could just want to change the voxel size but not the origin, so mandatory would be problematic in that case, right?

Reply:
How does this look (link, below)? The way I have it, there are defaults with a note in the description that they would be superseded by in_target. 

https://github.com/kesshijordan/nipype/blob/satra_fix_specs/nipype/interfaces/dtitk/utils.py

Reply:
Ah, yes, mandatory isn't necessary, then. I was going off of how this is currently written. My understanding now is: If target file is specified, array size, voxel size, and origin are defined by the properties of the target file and any explicit specifications are ignored; if it is not specified, then the properties of the input file are preserved, except where specified explicitly. So essentially none of these inputs is mandatory except `in_file`.

It's also unclear to me whether the positions are relevant. Typically, flagged arguments aren't required to be in a specific order. For the sake of simplicity, I'll assume they can be removed.

So, if that understanding is correct, then I might rewrite the input spec as follows (a couple comments within):

```Python
class TVResampleInputSpec(CommandLineInputSpec):
    in_file = File(desc="image to resample", exists=True,
                   mandatory=True, argstr="-in %s")
    out_file = traits.Str(desc='output path',
                          name_source="in_file", name_template="%s_resampled",
                          keep_extension=True, argstr="-out %s")
    target_file = File(desc='specs read from the target volume',
                       argstr="-target %s", xor=['array_size', 'voxel_size', 'origin'])
    align = traits.Str('center', argstr="-align %s")
    # This is missing an argstr. I'm not sure what it's supposed to be
    # Would also add a desc to make the options a little clearer
    interpolation = traits.Enum('LEI', 'EI')
    # Note the switch to ints and %d
    array_size = traits.Tuple((traits.Int(), traits.Int(), traits.Int()),
                           desc='resampled array size', xor=['target_file'],
                           argstr="-size %d %d %d")
    # Using %g will almost always give better command strings than %f
    voxel_size = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
                         desc='resampled voxel size', xor=['target_file'],
                         argstr="-vsize %g %g %g")
    origin = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
                          desc='xyz origin', xor=['target_file'],
                          argstr='-origin %g %g %g')
```

(Since you've done some renaming of these input fields, I've taken the liberty of renaming to names that I feel are a little more descriptive. Feel free to reject these changes if you prefer the ones you had.)

Note the `xor` arguments for `target_file`, `array_size`, `voxel_size`, and `origin`. That ensures that users aren't mistakenly passing parameters that will be ignored. (Again, I'm assuming that `array_size` should be included in this group. You can just remove its `xor` and remove it from `target_file`'s `xor` list, if I'm wrong.)

Reply:
Thanks, @effigies ! This is very helpful. The xor is a good idea to make sure the user is aware their inputs are ignored. I've fixed your other notes, here, and will apply them to other interfaces.